pandas/core/frame.py

```diff
@@ -9463,11 +9463,26 @@ def diff(self, periods: int = 1, axis: Axis = 0) -> DataFrame:
             periods = int(periods)
 
         axis = self._get_axis_number(axis)
+
+        def is_columns_bool_dtypes(self) -> bool:
```

Comment:
`(self.dtypes == np.bool_).all()`?

Reply:
Oh, thank you. I missed that `(self.dtypes == np.bool_).all()` could be used instead of `is_columns_bool_dtypes()`.